File: python/drivers/gdi_driver_stub.py
from printer_protocol import PrinterDriver


class GdiDriverStub(PrinterDriver):
    """Mock driver simulating graphical output (console only)."""

    device_origin = "top-left"
    device_y_direction = "down"
    dpi = 300.0

    def setup(self, name):
        print(f"[SETUP] {name}")

    def set_font(self, name, size):
        print(f"[FONT] {name} {size}")

    def set_alignment(self, align):
        print(f"[ALIGN] {align}")

    def set_direction(self, direction):
        print(f"[DIR] {direction}")

    def move_to(self, x, y):
        x_dev, y_dev = self.to_device_coords(x, y)
        print(f"[MOVE] {x_dev},{y_dev}")

    def draw_text(self, text):
        print(f"[TEXT] {text}")

    def draw_barcode(self, value, type, width, ratio, height, size):
        print(f"[BARCODE] {type} '{value}' {width}x{height}")

    def comment(self, text):
        print(f"[COMMENT] {text}")

    def print_feed(self):
        print("[PRINTFEED]")

    def get_dpi(self):
        return self.dpi

File: csharp/PrinterProtocol/GdiDriverStub.cs
using System;

namespace PrinterProtocol;

public class GdiDriverStub : IPrinterDriver
{
    private LayoutContext _context = new(0, 0, "mm", "bottom-left", "up", null);
    private double _dpi = 300.0;
    private const string DeviceOrigin = "top-left";
    private const string DeviceYDirection = "down";

    public void ConfigureLayout(LayoutContext context)
    {
        _context = context;
        if (context.Dpi.HasValue)
        {
            _dpi = context.Dpi.Value;
        }
    }

    public void Setup(string labelName) => Console.WriteLine($"[SETUP] {labelName}");

    public void SetFont(string name, double size) => Console.WriteLine($"[FONT] {name} {size}");

    public void SetAlignment(string align) => Console.WriteLine($"[ALIGN] {align}");

    public void SetDirection(string dir) => Console.WriteLine($"[DIR] {dir}");

    public void MoveTo(double x, double y)
    {
        var (xDevice, yDevice) = _context.ToDeviceCoords(DeviceOrigin, DeviceYDirection, x, y);
        Console.WriteLine($"[MOVE] {xDevice},{yDevice}");
    }

    public void DrawText(string text) => Console.WriteLine($"[TEXT] {text}");

    public void DrawBarcode(string value, string type, int width, int ratio, int height, int size) =>
        Console.WriteLine($"[BARCODE] {type} '{value}' {width}x{height}");

    public void Comment(string text) => Console.WriteLine($"[COMMENT] {text}");

    public void PrintFeed() => Console.WriteLine("[PRINTFEED]");

    public double GetDpi() => _dpi;
}
